src/Helpers/JWKFetcher.php

```diff
@@ -29,22 +35,31 @@ class JWKFetcher
      *
      * @var array
      */
-    private $guzzleOptions;
+    private $httpOptions;
+
+    /**
+     * JWKS URI to use, set in $httpOptions['jwks_uri'].
+     *
+     * @var array
+     */
+    private $jwksUri;
 
     /**
      * JWKFetcher constructor.
      *
-     * @param CacheInterface|null $cache         Cache handler or null for no caching.
-     * @param array               $guzzleOptions Options for the Guzzle HTTP client.
+     * @param CacheInterface|null $cache      Cache handler or null for no caching.
+     * @param array               $httOptions HTTP options, including Guzzle options.
      */
-    public function __construct(CacheInterface $cache = null, array $guzzleOptions = [])
+    public function __construct(CacheInterface $cache = null, array $httOptions = [])
     {
         if ($cache === null) {
             $cache = new NoCacheHandler();
         }
 
-        $this->cache         = $cache;
-        $this->guzzleOptions = $guzzleOptions;
+        $this->cache       = $cache;
+        $this->httpOptions = $httOptions;
+        $this->jwksUri     = $this->httpOptions['jwks_uri'] ?? null;
+        unset( $this->httpOptions['jwks_uri'] );
```

Comment:
It does mutate, ya. Best way I could see to allow this option to pass through without adding another parameter to the constructor. I probably don't need to remove that key here but it felt like the right thing to do since the other params are passed into the HTTP library.

Reply:
A "JWK Fetcher" needs to know where to fetch the JWK from (file, url, path, etc). That would be this URI value, and I think it's a core parameter for this class to work. I would add it as an extra parameter if that's possible 👍

The code you posted is a bit hacky. Is there anyway to avoid mutating the param on it? What about passing it as is (without removing it) to the http client, would that work? Typically, the input sanitation is part of a library. In this case, is up to the http client to decide what to do with that unrecognized value.

Reply:
what about calling the receiveid param just "options" and then inside, along with the jwks uri, storing an array of options for the http client?

Reply:
> A "JWK Fetcher" needs to know where to fetch the JWK from (file, url, path, etc). That would be this URI value, and I think it's a core parameter for this class to work.

Or the parameter on a method

> Is there anyway to avoid mutating the param on it? What about passing it as is (without removing it) to the http client, would that work?

Probably, yes? But it's not an option that needs to be passed there. 

> what about calling the receiveid param just "options" and then inside, along with the jwks uri, storing an array of options for the http client?

Would be great but would involve BC